src/luagit2/oid/lua_oid.c

```diff
@@ -0,0 +1,144 @@
+#include "lua_oid.h"
+
+int lua_git_oid_fromstr (lua_State *L) {
+	luagit2_oid *obj_id;
+	const char *string_value = luaL_checkstring(L, 1);
+
+	obj_id = (luagit2_oid *)lua_newuserdata(L, sizeof(*obj_id));
```

Comment:
I wonder if there are better ways to do this in a more Lua way instead of userdata.
Perhaps a regular table with a metatable would be best. Additionally, maybe the to_str method could be hidden from the api and called through the tostring metamethod. 
I think this is OK for the moment, but should be an improvement for the future!